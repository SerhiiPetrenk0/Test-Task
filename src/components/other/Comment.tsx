import React from 'react';
import { Card } from 'react-bootstrap';
import { StarBar } from './StarBar';
import { StyledCard } from '../../styled/other/Comment';

export const Comment = (props: { ListComment: any; }) => {
    const { ListComment } = props;
    return (
        <StyledCard>
            <Card.Body>
                <Card.Text>
                {ListComment.comments}
                </Card.Text>
                <Card.Subtitle>
                <StarBar count={ListComment.rating} />
                </Card.Subtitle>
            </Card.Body>
        </StyledCard>
    );
};
