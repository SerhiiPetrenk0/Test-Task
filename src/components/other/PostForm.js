import React , { useState } from 'react';
import { Form , Button } from 'react-bootstrap';
import { postCommentAPI } from '../../api';
import { colors } from '../../styled/globalStyled';
import { 
    StyledContainer,
    StyledFormGroup,
    StyledBsStarFill
} from '../../styled/other/PostForm';

export const PostForm = () => {
    const [formValue, setFormValue] = useState(
        {
            id: 'B0721KZEEE',
            rating: '',
            coments: ''
        }
    );
    const [currentValue, setCurrentValue] = useState(0);
    const [hoverValue, setHoverValue] = useState(undefined);
    const stars = Array(5).fill(0);
  
    const handleClick = value => {
      setCurrentValue(value);
      setFormValue({...formValue, rating:value});
    };
  
    const handleMouseOver = newHoverValue => setHoverValue(newHoverValue);

    const handleMouseLeave = () => setHoverValue(undefined);

    const handleTextarea = val => setFormValue({...formValue, coments:val});

    const handleSubmit = event => {
        event.preventDefault();
        postCommentAPI(formValue);
    };

    return(
        <StyledContainer>
            <Form onSubmit={handleSubmit}>
                <StyledFormGroup controlId="exampleForm.ControlInput1">
                    {stars.map((_, index) => (
                            <StyledBsStarFill
                            key={index}
                            size={24}
                            onClick={() => handleClick(index + 1)}
                            onMouseOver={() => handleMouseOver(index + 1)}
                            onMouseLeave={handleMouseLeave}
                            color={(hoverValue || currentValue) > index ? colors.starActive : colors.starDisable}
                            />
                    ))}
                </StyledFormGroup>
                <StyledFormGroup controlId="exampleForm.ControlTextarea1">
                    <Form.Label>comment</Form.Label>
                    <Form.Control as="textarea" rows={3} value={formValue.coments} onChange={event => handleTextarea(event.target.value)} />
                </StyledFormGroup>
                  <Button variant="primary" type="submit">
                        Submit
                  </Button>
            </Form>
        </StyledContainer>
    );
};
