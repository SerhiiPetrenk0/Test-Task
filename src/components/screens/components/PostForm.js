import React , { useState } from 'react';
import { Form , Button } from 'react-bootstrap';
import { postCommentAPI } from '../../../api';
import { 
    StyledContainer,
    StyledFormGroup,
    StyledBsStarFill
} from '../../../styled/screens/components/PostForm';

const colors = {
    orange: '#FFBA5A',
    grey: '#a9a9a9'
    
};

export const PostForm = () => {
    const [formValue, setFormValue ] = useState(
        {
            id: 'B0721KZEEE',
            rating: '',
            coments: ''
        }
    )
    const [currentValue, setCurrentValue] = useState(0);
    const [hoverValue, setHoverValue] = useState(undefined);
    const stars = Array(5).fill(0)
  
    const handleClick = value => {
      setCurrentValue(value)
      setFormValue({...formValue, rating:value})
    }
  
    const handleMouseOver = newHoverValue => setHoverValue(newHoverValue);

    const handleMouseLeave = () => setHoverValue(undefined);

    const handleTextarea = val => setFormValue({...formValue, coments:val});

    const handleSubmit = event => {
        event.preventDefault()
        postCommentAPI(formValue)
    }

    return(
        <StyledContainer>
            <Form onSubmit={handleSubmit}>
                <StyledFormGroup controlId="exampleForm.ControlInput1">
                    {stars.map((_, index) => (
                            <StyledBsStarFill
                            key={index}
                            size={24}
                            onClick={() => handleClick(index + 1)}
                            onMouseOver={() => handleMouseOver(index + 1)}
                            onMouseLeave={handleMouseLeave}
                            color={(hoverValue || currentValue) > index ? colors.orange : colors.grey}
                            />
                    ))}
                </StyledFormGroup>
                <StyledFormGroup controlId="exampleForm.ControlTextarea1">
                    <Form.Label>comment</Form.Label>
                    <Form.Control as="textarea" rows={3} value={formValue.coments} onChange={event => handleTextarea(event.target.value)} />
                </StyledFormGroup>
                  <Button variant="primary" type="submit">
                        Submit
                  </Button>
            </Form>
        </StyledContainer>
    )
}