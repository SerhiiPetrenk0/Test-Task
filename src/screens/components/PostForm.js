import React , { useState } from "react";
import { Form , Container, Button } from "react-bootstrap";
import { BsStarFill } from "react-icons/bs";

const colors = {
    orange: "#FFBA5A",
    grey: "#a9a9a9"
    
};

export const PostForm = () => {
    const [formValue, setFormValue ] = useState(
        {
            id: "B0721KZEEE",
            rating: "",
            coments: ""
        }
    )
    const [currentValue, setCurrentValue] = useState(0);
    const [hoverValue, setHoverValue] = useState(undefined);
    const stars = Array(5).fill(0)
  
    const handleClick = value => {
      setCurrentValue(value)
      setFormValue({...formValue, rating:value})
    }
  
    const handleMouseOver = newHoverValue => {
      setHoverValue(newHoverValue)
    };
  
    const handleMouseLeave = () => {
      setHoverValue(undefined)
    }
    const handleTextarea = (val) => {
        setFormValue({...formValue, coments:val})
    } 
    const handleSubmit = (event) => {
        event.preventDefault()
        fetch('http://demo8021751.mockable.io/comments', {
            method: 'POST',
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify(formValue)
        })
        .then((response) => response.json())
        .then((result) =>  result.result.push(formValue))


        
    }
    

    return(
        <Container className="mt-5 mb-4">
            <Form onSubmit={handleSubmit}>
                <Form.Group className="mb-3" controlId="exampleForm.ControlInput1">
                    {stars.map((_, index) => {
                        return (
                            <BsStarFill
                            key={index}
                            size={24}
                            onClick={() => handleClick(index + 1)}
                            onMouseOver={() => handleMouseOver(index + 1)}
                            onMouseLeave={handleMouseLeave}
                            color={(hoverValue || currentValue) > index ? colors.orange : colors.grey}
                            style={{
                                marginRight: 7,
                                cursor: "pointer"
                            }}
                            />
                        )
                    })}
                </Form.Group>
                <Form.Group className="mb-3" controlId="exampleForm.ControlTextarea1">
                    <Form.Label>comment</Form.Label>
                    <Form.Control as="textarea" rows={3} value={formValue.coments} onChange={event => handleTextarea(event.target.value)} />
                </Form.Group>
                  <Button variant="primary" type="submit">
                        Submit
                  </Button>
            </Form>
        </Container>
    )
}