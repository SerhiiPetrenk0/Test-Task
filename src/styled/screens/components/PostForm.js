import styled from 'styled-components'
import { BsStarFill } from 'react-icons/bs';
import { Container, Form } from 'react-bootstrap'

export const StyledContainer = styled(Container)`
    margin-top: 25px;
    margin-bottom: 40px
`;
export const StyledFormGroup = styled(Form.Group)`
    margin-bottom: 20px
`;
export const StyledBsStarFill = styled(BsStarFill)`
    margin-right: 7px;
    cursor: pointer
`;