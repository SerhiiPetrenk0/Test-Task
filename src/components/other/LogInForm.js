import React from "react"
import * as Yup from 'yup'
import { Form, Button, Modal } from "react-bootstrap"
import { useDispatch  } from "react-redux"
import { useFormik } from "formik"
import { StyledValidEmail } from "../../styled/other/LogInForm"
import { fillUser } from "../../redux/redusers/userInfoReduser"

export const LogInForm = props => {
    const { show, handleClose } = props
    const dispatch = useDispatch()
    const formik = useFormik({
      initialValues:{
        email: '',
        password: ''
      },
      validationSchema: Yup.object({
        email: Yup
        .string()
        .max(30, 'Max simbol 15))')
        .required('* this field is required')
        .matches(/^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/ , "Invalid email"),
        password: Yup
        .string()
        .required('passwordrequired')
        .matches(/^(?=.*[0-9])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{6,16}$/ , 'FALSE PASSWORD')
      }),
      onSubmit: values => {
        console.log(values)
        dispatch(fillUser(formik.values))
        handleClose()
      }
    })
    //почистити код / повиносити змінні / проблеми із змінною check / регулярки в змінні / 
      const emaillog = !!formik.errors.email ? formik.errors.email : "We'll never share your email with anyone else."
      const passwordlog = !!formik.errors.password ? formik.errors.password : "Your password must be 6 to 16 characters long, contain letters, numbers, special characters !@#$%^&*, And no spaces or emojis."
      const emailMassage = <StyledValidEmail check={!!formik.errors.email}>{emaillog}</StyledValidEmail>
      const passwordMassage = <StyledValidEmail check={!!formik.errors.password}>{passwordlog}</StyledValidEmail>
     console.log(!!formik.errors.password)
    return (
<>
  <Modal show={show} onHide={handleClose} backdrop="static" keyboard={false}>
    <Modal.Header closeButton>
      <Modal.Title>Modal title</Modal.Title>
    </Modal.Header>
    <Modal.Body>
      <Form onSubmit={formik.handleSubmit}>
        <Form.Group className="mb-3" controlId="formBasicEmail">
          <Form.Label>Email address</Form.Label>
          <Form.Control name="email" onChange={formik.handleChange} value={formik.values.email} type="email" placeholder="Enter email" />
          {emailMassage}
        </Form.Group>

        <Form.Group className="mb-3" controlId="formBasicPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control name="password" onChange={formik.handleChange} value={formik.values.password} type="password" placeholder="Password" />
          {passwordMassage}
        </Form.Group>

        <Button variant="primary" type="submit">
          Submit
        </Button>
      </Form>
    </Modal.Body>
  </Modal>
</>
    )
} 