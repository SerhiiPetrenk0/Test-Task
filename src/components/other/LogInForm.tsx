import React from 'react';
import { Form, Button, Modal } from 'react-bootstrap';
import { useDispatch  } from 'react-redux';
import { useFormik } from 'formik';

import { StyledValidEmail } from '../../styled/other/LogInForm';
import { fillUser } from '../../redux/ducks/userInfoDuck';
import { TypeFormikLogInForm, TypeUserinfo } from '../../interface';
import { YopLoginForm } from '../../validation';

interface TypeProps {
  show: boolean,
  handleClose: () => void
}

export const LogInForm: React.FC<TypeProps> = props => {
    const { show, handleClose } = props;
    const dispatch = useDispatch();
    const formik: TypeFormikLogInForm = useFormik({
      initialValues: {
        email: '',
        password: ''
      },
      validationSchema: YopLoginForm,
      onSubmit: (values: TypeUserinfo) => {
        dispatch(fillUser(formik.values))
        handleClose()
      }
    });
    
    const emailLog = !!formik.errors.email ?
      formik.errors.email :
      "We'll never share your email with anyone else.";
    const emailMassage: JSX.Element = (
      <StyledValidEmail check={formik.errors.email}>
        {emailLog}
      </StyledValidEmail>
    );

    const passwordLog = !!formik.errors.password ?
      formik.errors.password : 
      "Your password must be 6 to 16 characters long, contain letters, numbers, special characters !@#$%^&*, And no spaces or emojis."
    const passwordMassage: JSX.Element = (
      <StyledValidEmail check={formik.errors.password}>
        {passwordLog}
      </StyledValidEmail>
    );

    return (
      <>
        <Modal 
          show={show} 
          onHide={handleClose} 
          backdrop="static" 
          keyboard={false}
        >
          <Modal.Header closeButton>
            <Modal.Title>Registration</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            <Form onSubmit={formik.handleSubmit}>
              <Form.Group className="mb-3" controlId="formBasicEmail">
                <Form.Label>Email address</Form.Label>
                <Form.Control 
                  name="email" 
                  onChange={formik.handleChange} 
                  value={formik.values.email} 
                  type="email" 
                  placeholder="Enter email" 
                />
                {emailMassage}
              </Form.Group>

              <Form.Group className="mb-3" controlId="formBasicPassword">
                <Form.Label>Password</Form.Label>
                <Form.Control 
                  name="password" 
                  onChange={formik.handleChange} 
                  value={formik.values.password} 
                  type="password" 
                  placeholder="Password" 
                />
                {passwordMassage}
              </Form.Group>

              <Button variant="primary" type="submit">
                Submit
              </Button>
            </Form>
          </Modal.Body>
        </Modal>
      </>
    );
} ;
