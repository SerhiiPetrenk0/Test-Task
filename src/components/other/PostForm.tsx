import React , { useState } from 'react';
import { Form , Button } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { LogInForm } from './LogInForm';

import { formComment } from '../../redux/ducks/productsDuck';
import { colors } from '../../styled/globalStyled';
import { StyledContainer, StyledFormGroup, StyledBsStarFill } from '../../styled/other/PostForm';
import { TypeUserinfo, TypePostForm } from '../../interface';

export const PostForm:React.FC = () => {
    const [currentValue, setCurrentValue] = useState<number>(0);
    const [hoverValue, setHoverValue] = useState<undefined | number>(undefined);
    const [show, setShow] = useState<boolean>(false);
    const [formValue, setFormValue] = useState<TypePostForm>(
        {
            rating: '',
            comments: '',
        }
    );
    
    const stars: number[] = Array(5).fill(0);
    const dispatch = useDispatch();
    const userInfo: TypeUserinfo = useSelector((store:{userInfo:{userInfo: TypeUserinfo}}) => store.userInfo.userInfo);

    const handleMouseOver = (newHoverValue: React.SetStateAction<number | undefined>) => setHoverValue(newHoverValue);
    const handleMouseLeave = () => setHoverValue(undefined);
    const handleTextarea = (val: string) => setFormValue({...formValue, comments:val});
    const handleClose = () => setShow(false);
    const handleShow = () => setShow(true);
    const handleSubmit = (event: { preventDefault: () => void; }) => {
        event.preventDefault();
        if (!!userInfo.email) {
            dispatch(formComment(formValue));
            setFormValue({ ...formValue, comments:''});
            setCurrentValue(0);
        } else {
            handleShow()
        }
    };
    const handleClick = (value: React.SetStateAction<number>) => {
        setCurrentValue(value);
        setFormValue({...formValue, rating: value.toString()});
    };

    return (
        <>
            <LogInForm show={show} handleClose={handleClose} />
            <StyledContainer>
                <Form onSubmit={handleSubmit}>
                    <StyledFormGroup controlId="exampleForm.ControlInput1">
                        {stars.map((_, index) => (
                                <StyledBsStarFill
                                key={index}
                                size={24}
                                onClick={() => handleClick(index + 1)}
                                onMouseOver={() => handleMouseOver(index + 1)}
                                onMouseLeave={handleMouseLeave}
                                color={(hoverValue || currentValue) > index ? colors.starActive : colors.starDisable}
                                />
                        ))}
                    </StyledFormGroup>
                    <StyledFormGroup controlId="exampleForm.ControlTextarea1">
                        <Form.Label>comment</Form.Label>
                        <Form.Control as="textarea" rows={3} value={formValue.comments} onChange={event => handleTextarea(event.target.value)} />
                    </StyledFormGroup>
                    <Button variant="primary" type="submit">
                            Submit
                    </Button>
                </Form>
            </StyledContainer>
        </>
    );
};
