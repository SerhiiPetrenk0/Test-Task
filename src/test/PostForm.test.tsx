import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import * as reactRedux from 'react-redux'
import 'jsdom-global/register';

import { PostForm } from '../components/other/PostForm';

import store from '../redux';

configure({ adapter: new Adapter() });

describe('PostForm component', () => {
  beforeEach(() => {
    const dummyDispatch = jest.fn();
    const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
    const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
    useSelectorMock.mockReturnValue(store);
    useDispatchMock.mockReturnValue(dummyDispatch);
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  it('should match a snapshot (PostForm.test.tsx.snap)', () => {
    expect(toJson(mount(<PostForm />))).toMatchSnapshot();
  });
  
  it('should match a snapshot (PostForm.test.tsx.snap) after clikc button', () => {
    const component = shallow(<PostForm />);
    mount(<PostForm />).find('Button').find('button').simulate('click');
    expect(toJson(component)).toMatchSnapshot();
  });
});
