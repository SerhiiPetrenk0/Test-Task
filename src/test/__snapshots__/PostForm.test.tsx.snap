// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostForm component should match a snapshot (PostForm.test.tsx.snap) 1`] = `
<PostForm>
  <LogInForm
    handleClose={[Function]}
    show={false}
  >
    <Modal
      animation={true}
      autoFocus={true}
      backdrop="static"
      dialogAs={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      enforceFocus={true}
      keyboard={false}
      onHide={[Function]}
      restoreFocus={true}
      show={false}
    >
      <Modal
        autoFocus={true}
        backdrop="static"
        backdropTransition={[Function]}
        enforceFocus={true}
        keyboard={true}
        manager={
          BootstrapModalManager {
            "handleContainerOverflow": true,
            "isRTL": false,
            "modals": Array [],
            "ownerDocument": undefined,
          }
        }
        onEnter={[Function]}
        onEntering={[Function]}
        onEscapeKeyDown={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onHide={[Function]}
        renderBackdrop={[Function]}
        renderDialog={[Function]}
        restoreFocus={true}
        show={false}
        transition={[Function]}
      />
    </Modal>
  </LogInForm>
  <Styled(Container)>
    <Container
      className="sc-dkzDqf kRrPCR"
      fluid={false}
    >
      <div
        className="sc-dkzDqf kRrPCR container"
      >
        <Form
          onSubmit={[Function]}
        >
          <form
            className=""
            onSubmit={[Function]}
          >
            <Styled(FormGroup)
              controlId="exampleForm.ControlInput1"
            >
              <FormGroup
                className="sc-hKMtZM eVOESk"
                controlId="exampleForm.ControlInput1"
              >
                <div
                  className="sc-hKMtZM eVOESk"
                >
                  <Styled(BsStarFill)
                    color="#a9a9a9"
                    key="0"
                    onClick={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    size={24}
                  >
                    <BsStarFill
                      className="sc-eCYdqJ lmMuIV"
                      color="#a9a9a9"
                      onClick={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      size={24}
                    >
                      <IconBase
                        attr={
                          Object {
                            "fill": "currentColor",
                            "viewBox": "0 0 16 16",
                          }
                        }
                        className="sc-eCYdqJ lmMuIV"
                        color="#a9a9a9"
                        onClick={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        size={24}
                      >
                        <svg
                          className="sc-eCYdqJ lmMuIV"
                          color="#a9a9a9"
                          fill="currentColor"
                          height={24}
                          onClick={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#a9a9a9",
                            }
                          }
                          viewBox="0 0 16 16"
                          width={24}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                            key="0"
                          />
                        </svg>
                      </IconBase>
                    </BsStarFill>
                  </Styled(BsStarFill)>
                  <Styled(BsStarFill)
                    color="#a9a9a9"
                    key="1"
                    onClick={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    size={24}
                  >
                    <BsStarFill
                      className="sc-eCYdqJ lmMuIV"
                      color="#a9a9a9"
                      onClick={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      size={24}
                    >
                      <IconBase
                        attr={
                          Object {
                            "fill": "currentColor",
                            "viewBox": "0 0 16 16",
                          }
                        }
                        className="sc-eCYdqJ lmMuIV"
                        color="#a9a9a9"
                        onClick={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        size={24}
                      >
                        <svg
                          className="sc-eCYdqJ lmMuIV"
                          color="#a9a9a9"
                          fill="currentColor"
                          height={24}
                          onClick={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#a9a9a9",
                            }
                          }
                          viewBox="0 0 16 16"
                          width={24}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                            key="0"
                          />
                        </svg>
                      </IconBase>
                    </BsStarFill>
                  </Styled(BsStarFill)>
                  <Styled(BsStarFill)
                    color="#a9a9a9"
                    key="2"
                    onClick={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    size={24}
                  >
                    <BsStarFill
                      className="sc-eCYdqJ lmMuIV"
                      color="#a9a9a9"
                      onClick={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      size={24}
                    >
                      <IconBase
                        attr={
                          Object {
                            "fill": "currentColor",
                            "viewBox": "0 0 16 16",
                          }
                        }
                        className="sc-eCYdqJ lmMuIV"
                        color="#a9a9a9"
                        onClick={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        size={24}
                      >
                        <svg
                          className="sc-eCYdqJ lmMuIV"
                          color="#a9a9a9"
                          fill="currentColor"
                          height={24}
                          onClick={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#a9a9a9",
                            }
                          }
                          viewBox="0 0 16 16"
                          width={24}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                            key="0"
                          />
                        </svg>
                      </IconBase>
                    </BsStarFill>
                  </Styled(BsStarFill)>
                  <Styled(BsStarFill)
                    color="#a9a9a9"
                    key="3"
                    onClick={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    size={24}
                  >
                    <BsStarFill
                      className="sc-eCYdqJ lmMuIV"
                      color="#a9a9a9"
                      onClick={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      size={24}
                    >
                      <IconBase
                        attr={
                          Object {
                            "fill": "currentColor",
                            "viewBox": "0 0 16 16",
                          }
                        }
                        className="sc-eCYdqJ lmMuIV"
                        color="#a9a9a9"
                        onClick={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        size={24}
                      >
                        <svg
                          className="sc-eCYdqJ lmMuIV"
                          color="#a9a9a9"
                          fill="currentColor"
                          height={24}
                          onClick={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#a9a9a9",
                            }
                          }
                          viewBox="0 0 16 16"
                          width={24}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                            key="0"
                          />
                        </svg>
                      </IconBase>
                    </BsStarFill>
                  </Styled(BsStarFill)>
                  <Styled(BsStarFill)
                    color="#a9a9a9"
                    key="4"
                    onClick={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    size={24}
                  >
                    <BsStarFill
                      className="sc-eCYdqJ lmMuIV"
                      color="#a9a9a9"
                      onClick={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      size={24}
                    >
                      <IconBase
                        attr={
                          Object {
                            "fill": "currentColor",
                            "viewBox": "0 0 16 16",
                          }
                        }
                        className="sc-eCYdqJ lmMuIV"
                        color="#a9a9a9"
                        onClick={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        size={24}
                      >
                        <svg
                          className="sc-eCYdqJ lmMuIV"
                          color="#a9a9a9"
                          fill="currentColor"
                          height={24}
                          onClick={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": "#a9a9a9",
                            }
                          }
                          viewBox="0 0 16 16"
                          width={24}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.282.95l-3.522 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
                            key="0"
                          />
                        </svg>
                      </IconBase>
                    </BsStarFill>
                  </Styled(BsStarFill)>
                </div>
              </FormGroup>
            </Styled(FormGroup)>
            <Styled(FormGroup)
              controlId="exampleForm.ControlTextarea1"
            >
              <FormGroup
                className="sc-hKMtZM eVOESk"
                controlId="exampleForm.ControlTextarea1"
              >
                <div
                  className="sc-hKMtZM eVOESk"
                >
                  <FormLabel
                    column={false}
                    visuallyHidden={false}
                  >
                    <label
                      className="form-label"
                      htmlFor="exampleForm.ControlTextarea1"
                    >
                      comment
                    </label>
                  </FormLabel>
                  <FormControl
                    as="textarea"
                    onChange={[Function]}
                    rows={3}
                    value=""
                  >
                    <textarea
                      className="form-control"
                      id="exampleForm.ControlTextarea1"
                      onChange={[Function]}
                      rows={3}
                      value=""
                    />
                  </FormControl>
                </div>
              </FormGroup>
            </Styled(FormGroup)>
            <Button
              active={false}
              disabled={false}
              type="submit"
              variant="primary"
            >
              <button
                className="btn btn-primary"
                disabled={false}
                type="submit"
              >
                Submit
              </button>
            </Button>
          </form>
        </Form>
      </div>
    </Container>
  </Styled(Container)>
</PostForm>
`;

exports[`PostForm component should match a snapshot (PostForm.test.tsx.snap) after clikc button 1`] = `
<Fragment>
  <LogInForm
    handleClose={[Function]}
    show={false}
  />
  <Styled(Container)>
    <Form
      onSubmit={[Function]}
    >
      <Styled(FormGroup)
        controlId="exampleForm.ControlInput1"
      >
        <Styled(BsStarFill)
          color="#a9a9a9"
          key="0"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          size={24}
        />
        <Styled(BsStarFill)
          color="#a9a9a9"
          key="1"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          size={24}
        />
        <Styled(BsStarFill)
          color="#a9a9a9"
          key="2"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          size={24}
        />
        <Styled(BsStarFill)
          color="#a9a9a9"
          key="3"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          size={24}
        />
        <Styled(BsStarFill)
          color="#a9a9a9"
          key="4"
          onClick={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          size={24}
        />
      </Styled(FormGroup)>
      <Styled(FormGroup)
        controlId="exampleForm.ControlTextarea1"
      >
        <FormLabel
          column={false}
          visuallyHidden={false}
        >
          comment
        </FormLabel>
        <FormControl
          as="textarea"
          onChange={[Function]}
          rows={3}
          value=""
        />
      </Styled(FormGroup)>
      <Button
        active={false}
        disabled={false}
        type="submit"
        variant="primary"
      >
        Submit
      </Button>
    </Form>
  </Styled(Container)>
</Fragment>
`;
